javascript-practice/src/js/controllers/product-form.controller.js

```diff
@@ -1,32 +1,47 @@
 import { getElementById, generateErrorMessages } from '../utils/dom';
 import validateForm from '../utils/validateForm';
+import findRoute from '../utils/findRoute';
 
-export default class ProductAddController {
-  constructor(view, service) {
+export default class ProductFormController {
+  constructor(view, service, action) {
     this.view = view;
     this.service = service;
+    this.action = action;
```

Comment:
Seems like you got 2 of action edit and add through your code, so pls  help to create it as constant to make it more consistent.